### PR TITLE
Allow medic conf to progress even when it can not get the latest update data

### PR DESCRIPTION
**Pull request by nomulex**
Medic-conf was choking and could not progress if it got an error for example this one below
`ERROR SyntaxError: Unexpected end of JSON input at JSON.parse (<anonymous>) at request.get.then.res (/Users/kennethatria/.nvm/versions/node/v8.11.0/lib/node_modules/medic-conf/src/lib/check-for-updates.js:14:20) at <anonymous> at process._tickCallback (internal/process/next_tick.js:188:7)`

So I created a minor fix.

### CODE REVIEW

**Comment by nomulex** [comment]
Looks like for slow networks , it was  downloading the JSON half way. Here's the error it was throwing. 
`ERROR SyntaxError: Unexpected end of JSON input at JSON.parse (<anonymous>) at request.get.then.res (//.nvm/versions/node/v8.11.0/lib/node_modules/medic-conf/src/lib/check-for-updates.js:14:20) at <anonymous> at process._tickCallback (internal/process/next_tick.js:188:7)`

File: src/lib/check-for-updates.js [hunk 1]
```diff
@@ -6,31 +6,33 @@ module.exports = (options) => {
...
-	} else throw err;
-      })
-      ;
+      }
+    })
+    .catch(err => {
+      if (options.nonFatal && err.cause && err.cause.code === 'ENOTFOUND') {
+        warn('Could not check NPM for updates.  You may be offline.');
+      } else {
+        warn(`Could not check NPM for updates. Error:${err.message}`);
```
Comment (SCdF) [inline src/lib/check-for-updates.js]:
```suggestion
        warn(`Could not check NPM for updates. Error: ${err.message}`);
```

**Comment by SCdF** [comment]
Nice change @nomulex, good job!

**Comment by nomulex** [comment]
Thanks for the help @SCdF !

**Pull request merged**